### PR TITLE
3325715: update php version

### PR DESCRIPTION
**Pull request by cbfannin**
composer.json, .lando.yml, circle.config updated to use php 8.1.

DO issue: https://www.drupal.org/project/sous/issues/3325715

**NOTES**
I ran PHP compatibility and the following were warnings that I have not found solutions for yet.
`Since PHP 7.0, functions inspecting arguments, like func_get_args(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter \$callback\" was used on line 705.`
`Since PHP 7.0, functions inspecting arguments, like debug_backtrace(), no longer report the original value as passed to a parameter, but will instead provide the current value. The parameter \$error_level\" was used on line 59.`

### CODE REVIEW

**Comment by cbfannin** [comment]
@ccjjmartin I reviewed circleci config and saw no reference of the php version. I think this means it is defaulting to PHP 7.4 though. I don't know enough about setting up CircleCi yet so need help with this.

**Comment by cbfannin** [comment]
OK I think this is ready. I believe I got php working in CircleCI based on thes docs I was able to add the correct image: https://circleci.com/developer/images/image/cimg/php

File: README.md [hunk 1]
```diff
@@ -133,7 +133,7 @@ composer create-project fourkitchens/sous-drupal-project:dev-[branch-name] PROJE
 ## Contributing without create-project or creating a project with a custom theme
 1. Clone this repository
 2. Checkout the 4.x-beta branch (if it hasn't been released yet)
-3. `composer install` (you must have greater than PHP 7.4 installed locally)
+3. `composer install` (you must have greater than PHP 8.1 installed locally)
```
Comment (ccjjmartin) [inline README.md:136]: @cbfannin Can we switch this to `lando composer install`?

**Review by ccjjmartin — APPROVED** [review]
@cbfannin One comment on the code, after that is updated code wise this looks good to me.  @birchy82 Can you test the functional on this one?

File: README.md (hunk 1 above)
Comment (cbfannin) [reply README.md]: Fixed. Its all your's @birchy82

Comment (ccjjmartin) [reply README.md]: @cbfannin My bad, can we remove the extra comment text too, people don't have to have PHP8.1 installed locally if they are using lando.  I forgot to put that in my previous comment.

Comment (cbfannin) [reply README.md]: Fixed.

Comment (ccjjmartin) [reply README.md]: Code looks great, thanks 👍

**Review by edwtorba — APPROVED** [review]
@cbfannin :shipit:

**Pull request merged**